revscoring/scoring/statistics/classification/classification.py

```diff
@@ -0,0 +1,143 @@
+        self.decision_key = decision_key
+        self.labels = labels
+        self.population_rates = population_rates or {}
+
+    def fit(self, score_labels):
+        """
+        Fit to scores and labels.
+
+        :Parameters:
+            score_labels : [( `dict`, `mixed` )]
+                A collection of scores-label pairs generated using
+                :class:`revscoring.Model.score`.  Note that fitting is usually
+                done using data withheld during model training
+        """
+        super().fit(score_labels)
+        if self.labels is None:
+            self.labels = sorted(set(l for s, l in score_labels))
+
+        self['counts'] = Counts(self.labels, score_labels, self.prediction_key)
+        self['rates'] = Rates(self['counts'],
+                              population_rates=self.population_rates)
+
+        self.label_stats = OrderedDict()
+        for label in self.labels:
+            self.label_stats[label] = ScaledPredictionStatistics(
+                y_preds=[s[self.prediction_key] == label
+                         for s, l in score_labels],
+                y_trues=[l == label for s, l in score_labels],
+                population_rate=self.population_rates.get(label))
+
+        self['match_rate'] = \
+            MicroMacroStats(self.label_stats, 'match_rate')
+        self['filter_rate'] = \
+            MicroMacroStats(self.label_stats, 'filter_rate')
+        self['recall'] = \
+            MicroMacroStats(self.label_stats, 'recall')
+        self['!recall'] = \
+            MicroMacroStats(self.label_stats, '!recall')
+        self['precision'] = \
+            MicroMacroStats(self.label_stats, 'precision')
+        self['!precision'] = \
+            MicroMacroStats(self.label_stats, '!precision')
+        self['f1'] = \
+            MicroMacroStats(self.label_stats, 'f1')
+        self['!f1'] = \
+            MicroMacroStats(self.label_stats, '!f1')
+        self['accuracy'] = \
+            MicroMacroStats(self.label_stats, 'accuracy')
+        self['fpr'] = \
+            MicroMacroStats(self.label_stats, 'fpr')
+
+        if self.decision_key is not None:
+            self.label_thresholds = OrderedDict()
+            for label in self.labels:
+                self.label_thresholds[label] = ScaledThresholdStatistics(
+                    [s[self.decision_key][label] for s, l in score_labels],
+                    [l == label for s, l in score_labels],
+                    population_rate=self.population_rates.get(label))
+
+            self['roc_auc'] = \
+                MicroMacroStats(self.label_thresholds, 'roc_auc')
+            self['pr_auc'] = \
+                MicroMacroStats(self.label_thresholds, 'pr_auc')
+
+    def __getitem__(self, key):
+        if key in self:
+            return super().__getitem__(key)
+        elif hasattr(self, "label_thresholds"):
+            #  This handles the case where a 'key' is a threshold optimization
```

Comment:
In this case, "key" can only be one of the items in self or a threshold optimization.  It's up to the user of this api whether or not they want to look inside of MacroMicroStats.  

I suppose we could try to just construct the threshold optimization here and report a better error.